tools/ci_build/github/azure-pipelines/templates/c-api-artifacts-package-and-publish-steps-windows.yml

```diff
@@ -36,6 +36,8 @@ steps:
 
           echo "Directories created"            
           copy $(Build.BinariesDirectory)\${{parameters.buildConfig}}\${{parameters.buildConfig}}\onnxruntime.dll $(Build.BinariesDirectory)\${{parameters.artifactName}}\lib
+          copy $(Build.BinariesDirectory)\${{parameters.buildConfig}}\${{parameters.buildConfig}}\onnxruntime_providers_shared.dll $(Build.BinariesDirectory)\${{parameters.artifactName}}\lib
```

Comment:
I don't think the copy can be done unconditionally?
I think this template is used by both cpu and gpu packaging?

Reply:
I thought that originally too, but it copies over the cuda_provider_factory.h unconditionally here. The non GPU packages get built just fine with this change.

Reply:
cuda_provider_factory.h exists though. 
I guess the copy for azp doesn't fail the job if the file doesn't exist.
see: https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=158281&view=logs&j=ec0bcaa2-12c5-5401-2f4d-3d4062704c3c&t=ce54dd27-dbd7-584b-5419-6508c10b37b4&l=15

Reply:
I also thought we should change it to *.dll to make it catch all provider libraries built if we build them, but I noticed my build also has 'custom_op_library.dll' and 'test_execution_provider.dll'. I'll check with Changming

Reply:
it could be onnxruntime_providers_*.dll then ?